src/groundlight/client.py

```diff
@@ -266,13 +266,15 @@ def create_detector(  # noqa: PLR0913
 
         :return: Detector
         """
-        detector_creation_input = DetectorCreationInput(name=name, query=query)
-        if confidence_threshold is not None:
-            detector_creation_input.confidence_threshold = confidence_threshold
-        if pipeline_config is not None:
-            detector_creation_input.pipeline_config = pipeline_config
+        detector_creation_input = DetectorCreationInputRequest(
+            name=name,
+            query=query,
+            pipeline_config=pipeline_config,
+        )
         if metadata is not None:
             detector_creation_input.metadata = str(url_encode_dict(metadata, name="metadata", size_limit_bytes=1024))
+        if confidence_threshold:
```

Comment:
Was there a reason to reorder this? Not saying you need to change it just wondering if Im missing something

Reply:
No, I think I just happened to move it down while I was moving the pipeline_config default into the constructor